ipn/ipnlocal/local.go

```diff
@@ -204,10 +204,11 @@ type LocalBackend struct {
 	capTailnetLock bool // whether netMap contains the tailnet lock capability
 	// hostinfo is mutated in-place while mu is held.
 	hostinfo         *tailcfg.Hostinfo
-	netMap           *netmap.NetworkMap     // not mutated in place once set (except for Peers slice)
+	netMap           *netmap.NetworkMap                  // not mutated in place once set (except for Peers slice)
+	peers            map[tailcfg.NodeID]tailcfg.NodeView // mutated in place while mu is held; map values may escape LocalBackend, map itself may not
```

Comment:
you now have two different lists of Peers in LocalBackend which go out of sync when you receive the first delta netmap